dbms/src/DataTypes/DataTypeDomainBool.cpp

```diff
@@ -0,0 +1,93 @@
+#include <Columns/ColumnsNumber.h>
+#include <Common/Exception.h>
+#include <DataTypes/DataTypeDomainWithSimpleSerialization.h>
+#include <DataTypes/DataTypeFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <IO/ReadBuffer.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+    extern const int CANNOT_PARSE_DOMAIN_VALUE_FROM_STRING;
+}
+
+namespace
+{
+
+class DataTypeDomainBool : public DataTypeDomainWithSimpleSerialization
```

Comment:
`final`

For devirtualization in forwarded virtual calls like `serializeTextJSON`.